packages/eas-cli/src/commands/update/configure.ts

```diff
@@ -7,13 +7,21 @@ import Log, { learnMore } from '../../log';
 import { findProjectRootAsync, getProjectIdAsync } from '../../project/projectUtils';
 import { resolveWorkflowAsync } from '../../project/workflow';
 
-const EAS_UPDATE_URL = 'https://u.expo.dev';
+const EAS_UPDATE_URL_PROD = 'https://u.expo.dev';
+const EAS_UPDATE_URL_STAGING = 'https://staging-u.expo.dev';
+const EAS_UPDATE_URL_LOCAL = 'http://localhost:3000';
 const DEFAULT_MANAGED_RUNTIME_VERSION = { policy: 'sdkVersion' } as const;
 const DEFAULT_BARE_RUNTIME_VERSION = '1.0.0';
 
 export async function getEASUpdateURLAsync(exp: ExpoConfig): Promise<string> {
   const projectId = await getProjectIdAsync(exp);
-  return new URL(projectId, EAS_UPDATE_URL).href;
+  if (process.env.EXPO_LOCAL) {
+    return new URL(`expo-updates/${projectId}`, EAS_UPDATE_URL_LOCAL).href;
+  } else if (process.env.EXPO_STAGING) {
+    return new URL(projectId, EAS_UPDATE_URL_STAGING).href;
+  } else {
+    return new URL(projectId, EAS_UPDATE_URL_PROD).href;
+  }
```

Comment:
Move this logic to https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/api.ts